### ISSUE TITLE
Reload .env after updates fail

### ISSUE THREAD

**Issue by vbakke** [gaborcsardi/dotenv#15]
Steps to reproduce:
* Open Rstudio with a project using dotenv
* Run the R script that uses `library(dotenv)`
* Change a variable in the `.env` file
* Rerun the script

The `Sys.getenv("TEST_VAR")` still reports the _old_ value of `TEST_VAR`.

**Comment by vbakke** [comment]
A workaround I've found is to manually run `detach("package:dotenv", unload=TRUE)` every time I update my .env file.

**Comment by timoruohomaki** [comment]
> A workaround I've found is to manually run `detach("package:dotenv", unload=TRUE)` every time I update my .env file.

Thanks, that solved it for me as well. Fortunately there are not that many updates but still.

**Comment by gaborcsardi** [comment]
Maybe dotenv could unload itself, so the repeated `library()` call would load it again, so it could load the updated env vars.

**Comment by gaborcsardi** [comment]
> Maybe dotenv could unload itself, so the repeated `library()` call would load it again, so it could load the updated env vars.

This is actually pretty difficult, so I won't do it. 

However, a good workaround is to use 
```
dotenv::load_dot_env()
```
in your script. Or, if you want to condition it on the existence of `.env`, then:
```
if (file.exists(".env")) dotenv::load_dot_env()
```

With this, the script will re-load `.env` every time it runs.

**Comment by vbakke** [comment]
Yes, running `dotenv::load_dot_env()` explicitley is waht I ended up doing as well. 

If unloading is difficult, /which I presume is it, I recommend changing the 'Usage' in the README to:

```
library(dotenv)
dotenv::load_dot_env()
```

Then I think this issue is solved.   :)
